### ISSUE TITLE
Lint/RedundantDirGlobSort is not safe and should not be applied for `Dir` with multiple string arguments

### ISSUE THREAD

**Issue by syakovyn** [rubocop/rubocop#11074]
Lint/RedundantDirGlobSort cop warns and autocorrects code like `Dir[*paths].sort` to be replaced with `Dir[*paths]`. However, in case of multiple paths the result of above 2 expressions differ.

--------

## Expected behavior

Do not warn about using `sort` in `Dir[*paths].sort` as being redundant.

## Actual behavior

Rubocop warns about using `sort` in `Dir[*paths].sort` and autocorrects it.

## Steps to reproduce the problem

I was able to reproduce the issue using the following setup:
1. Create 2 files in the working directory in the following order: `tmp/test/b/a.txt` and `tmp/test/a/c.txt`.
2. Run irb for ruby `ruby 3.0.4p208 (2022-04-12 revision 3fa771dded) [arm64-darwin21]`
3. Then `Dir['./tmp/test/b/*.txt', './tmp/test/a/*.txt'].sort` gives `["./tmp/test/a/c.txt", "./tmp/test/b/a.txt"]` but `Dir['./tmp/test/b/*.txt', './tmp/test/a/*.txt']` gives `["./tmp/test/b/a.txt", "./tmp/test/a/c.txt"]`

As you can see the order of files differ so this cop cannot be used to autocorrect or even warn in case of multiple paths passed to `Dir[]`.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.0.4) [arm64-darwin21]
  - rubocop-performance 1.15.0
  - rubocop-rails 2.16.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.13.1
```
